source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+
+Researching the changes in Rails a bit, we find ourselves with a regression inserted in [this commit](https://github.com/rails/rails/commit/491013e06d0ad4a296cc23be6c4a48bb0a98106f). This regression isn’t occurring because of the hidden field having been inserted via the radio buttons, but rather it appears because we are reusing all of the code used by the checkboxes, which need to send an Array of elements and not a String.
+In [this line](https://github.com/rails/rails/commit/491013e06d0ad4a296cc23be6c4a48bb0a98106f#diff-ad5a6748d86270f717368d13026b475bL44) specifically, note the [] at the end of the String – they were there and this only makes sense for the checkboxes. That code was moved [here](https://github.com/rails/rails/commit/491013e06d0ad4a296cc23be6c4a48bb0a98106f#diff-f881ed71156b4aa73c4604ca14b074d9R104) and that method is now utilized by both the radio buttons and the checkboxes. That’s where the error is – the radio buttons shouldn’t have [] at the end.
+
+It was necessary to fix the problem in Rails and upload it to the master :). [Here's the fix](https://github.com/rails/rails/commit/7e583b73c4e7ddad2241b2ca05b3c16a5fdf0cc6).
+With that fix in the master, we update the master reference that we’re using in our application simply by executing `bundle update rails`, running the tests and problem solved.
+
+One less problem, but there are still more.
+
+
+## Problem 3 - Model enum attribute not saved
+
+`bin/rails test` gives us ...
+
+```
+Expected to find text "Expense" in "Groceries Edit Delete".
+```
+
+In this test, the Category model, which has a name (in the test the value is Groceries) and a kind with the possible values `income`, `expense` and `any`, comes into play.
+
+The error suggests that the Category is not able to persist the kind value. We quickly try to save a Category with kind expense from the console ...
+
+```
+irb(main):001:0> Category.create!(name: 'Groceries', kind: 'expense')
+=> #<Category id: 1, name: "Groceries", kind: nil, created_at: "2016-01-20 19:29:35", updated_at: "2016-01-20 19:29:35">
+```
+
+There’s where the error comes from, kind isn't saving.
+
+In our case, kind is an enum in the Category model and is represented as a PostgreSQL enum. In order for this combination to work correctly, we need to “hack” the mapping that Active Record creates by default for the enums since, by default, they are returned as integers and we would need Strings for PostgreSQL. Hence, the enums declaration must be something like ... `enum kind: { expense: 'expense', income: 'income', any: 'any' }`. With that declaration, the enums stop using the default mapping to integers and each declared key is mapped to the String that we put in the Hash. This syntax functions correctly in Rails 4.2 but, for some reason, we are not able to save the enums in Rails 5.0.
```

Comment:
`we are not able to save the enums in Rails 5.0` => `it's not working correctly in Rails 5.0`